wgpu-hal/src/dx12/descriptor.rs

```diff
@@ -38,32 +40,30 @@ pub(super) struct GeneralHeap {
 
 impl GeneralHeap {
     pub(super) fn new(
-        device: d3d12::Device,
-        ty: d3d12::DescriptorHeapType,
+        device: &Direct3D12::ID3D12Device,
+        ty: Direct3D12::D3D12_DESCRIPTOR_HEAP_TYPE,
         total_handles: u64,
     ) -> Result<Self, crate::DeviceError> {
         let raw = {
             profiling::scope!("ID3D12Device::CreateDescriptorHeap");
-            device
-                .create_descriptor_heap(
-                    total_handles as u32,
-                    ty,
-                    d3d12::DescriptorHeapFlags::SHADER_VISIBLE,
-                    0,
-                )
+            let desc = Direct3D12::D3D12_DESCRIPTOR_HEAP_DESC {
+                Type: ty,
+                NumDescriptors: total_handles as u32,
+                Flags: Direct3D12::D3D12_DESCRIPTOR_HEAP_FLAG_SHADER_VISIBLE,
+                NodeMask: 0,
+            };
+            unsafe { device.CreateDescriptorHeap::<Direct3D12::ID3D12DescriptorHeap>(&desc) }
                 .into_device_result("Descriptor heap creation")?
         };
 
-        null_comptr_check(&raw)?;
-
         Ok(Self {
             raw: raw.clone(),
```

Comment:
No need to `clone()` (raise the refcount) if we construct `DualHandle` before `Self`.